### ISSUE TITLE
CoreCLR support

### ISSUE THREAD

**Issue by schotime** [mgholam/fastJSON#24]
Is this in the works? or thought about?

**Comment by mgholam** [comment]
I haven't looked into it yet, if CoreCLR supports dynamic methods and reflection emit, it should work.

**Comment by schotime** [comment]
It does. I'm using them in NPoco as well.

**Comment by schotime** [comment]
Hey @mgholam, I got it compiling on CoreCLR. I've currently embedded it in my NPoco repo with the following commit https://github.com/schotime/NPoco/commit/49c4fe6daa208c7cfb840c29c1258754211c692e.

The changes were this.
1. Most of the reflection stuff justed needed a GetTypeInfo() call on the type.
2. I don't support Silverlight atm so I just changed these compiler directives to be the dnxcore50 (this will change when RC2 is released)
3. Silverlight didn't support HashTable, but coreclr does so I changed that slightly. eg. https://github.com/schotime/NPoco/commit/49c4fe6daa208c7cfb840c29c1258754211c692e#diff-c6926a1bdeb07852ff5063c44e896234R222

Pretty sure they were the main ones.
Hope this helps.
